tests/integration/high_availability/test_self_healing.py

```diff
@@ -257,3 +257,65 @@ async def test_graceful_crash_of_primary(ops_test: OpsTest, continuous_writes) -
         for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(10)):
             with attempt:
                 assert ensure_all_units_continuous_writes_incrementing(ops_test)
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.self_healing_tests
+async def test_network_cut_affecting_an_instance(ops_test: OpsTest, continuous_writes) -> None:
+    """Test for a network cut affecting an instance."""
+    mysql_application_name, _ = await high_availability_test_setup(ops_test)
+
+    await ensure_all_units_continuous_writes_incrementing(ops_test)
+
+    deploy_chaos_mesh(ops_test, ops_test.model.info.name)
```

Comment:
This could be factored out as a fixture or a setup/teardown, so that `destroy_chaos_mesh()` is not called only if the test is successful.

Reply:
Great suggestion! Addressed in be15a67bd659c981a1705d287ef1d10a9e930f38